cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -575,40 +575,28 @@ private MDSKey getNamespaceKey(@Nullable String name) {
     return builder.build();
   }
 
-
   public void updateWorkflowToken(ProgramRunId workflowRunId, WorkflowToken workflowToken) throws NotFoundException {
-    RunRecordMeta runRecordMeta = getUnfinishedRun(workflowRunId.getParent().toId(), TYPE_RUN_RECORD_STARTED,
-                                                   workflowRunId.getRun());
-    if (runRecordMeta == null) {
-      // Even if a run is suspended, the currently running node runs to completion.
-      // This node also updates the workflow token at the end.
-      // For such a scenario, check for a suspended run record too.
-      runRecordMeta = getUnfinishedRun(workflowRunId.getParent().toId(), TYPE_RUN_RECORD_SUSPENDED,
-                                       workflowRunId.getRun());
-      if (runRecordMeta == null) {
-        throw new NotFoundException(workflowRunId);
-      }
-    }
-    Map<String, String> propertiesToUpdate = runRecordMeta.getProperties();
-    propertiesToUpdate.put(WORKFLOW_TOKEN_PROPERTY_KEY, GSON.toJson(workflowToken));
+    // Workflow token will be stored with following key:
+    // [wft][namespace][app][WORKFLOW][workflowName][workflowRun]
```

Comment:
Is ``[WORKFLOW]`` necessary? There shouldn't be workflow tokens for other program types right? Shouldn't ``namespace.app.workflowname.workflowrun`` uniquely identify it? same question for node states as well. 

Or do we want to keep it consistent with the other keys - not sure if that is necessary.